src/compiler/interface.jl

```diff
@@ -171,6 +173,39 @@ function copy!(x::AbstractVector,  gs::Grads)
   x
 end
 
+# # Gradient Algebra: unary 
+# for op in (:+, :-)
+#   @eval broadcasted(::typeof($op), gs::Grads) = map($op, gs)
+# end
+
+# # Gradient Algebra: binary 
+# for op in (:+, :-)
+#   @eval broadcasted(::typeof($op), gs1::Grads, gs2::Grads) = map($op, gs1, gs2)
+# end
+
+broadcasted(f, gss::Grads...) = map(f, gss...)
+broadcasted(f, gs1::Grads, xs...) = map(f, gs1, xs...)
+broadcasted(::typeof(*), a::Number, gs::Grads) = map(*, gs, a)
+
+function Base.map(f, gs1::Grads, gss::Grads...)
+  @assert all(issetequal(gs1.params, gs.params) for gs in gss)
+  grads = IdDict{Any,Any}()
+  ps = Params(gs1.params)
+  for p in ps
```

Comment:
```suggestion
  for p in keys(gs1)
```

Reply:
I need `ps` to construct the output Grads object